tasks_management/schema.py

```diff
@@ -0,0 +1,72 @@
+import graphene
+
+from django.contrib.auth.models import AnonymousUser
+from django.db.models import Q
+
+from core.schema import OrderedDjangoFilterConnectionField
+from core.utils import append_validity_filter
+from tasks_management.apps import TasksManagementConfig
+from tasks_management.gql_mutations import CreateTaskGroupMutation
+from tasks_management.gql_queries import TaskGroupGQLType, TaskExecutorGQLType
+from tasks_management.models import TaskGroup, TaskExecutor
+
+import graphene_django_optimizer as gql_optimizer
+
+
+class Query(graphene.ObjectType):
+    module_name = "tasks_management"
+
+    task_group = OrderedDjangoFilterConnectionField(
+        TaskGroupGQLType,
+        orderBy=graphene.List(of_type=graphene.String),
+        dateValidFrom__Gte=graphene.DateTime(),
+        dateValidTo__Lte=graphene.DateTime(),
+        client_mutation_id=graphene.String(),
+    )
+    task_executor = OrderedDjangoFilterConnectionField(
+        TaskExecutorGQLType,
+        orderBy=graphene.List(of_type=graphene.String),
+        dateValidFrom__Gte=graphene.DateTime(),
+        dateValidTo__Lte=graphene.DateTime(),
+        client_mutation_id=graphene.String(),
+    )
+
+    def resolve_task_group(self, info, **kwargs):
+        filters = append_validity_filter(**kwargs)
+
+        client_mutation_id = kwargs.get("client_mutation_id", None)
+        if client_mutation_id:
+            filters.append(Q(mutations__mutation__client_mutation_id=client_mutation_id))
+
+        Query._check_permissions(
+            info.context.user,
+            TasksManagementConfig.gql_task_group_search_perms
+        )
+        query = TaskGroup.objects.filter(*filters)
+        return gql_optimizer.query(query, info)
+
+    def resolve_task_executor(self, info, **kwargs):
+        filters = append_validity_filter(**kwargs)
+
+        client_mutation_id = kwargs.get("client_mutation_id")
+        if client_mutation_id:
+            filters.append(Q(mutations__mutation__client_mutation_id=client_mutation_id))
+
+        task_group_id = kwargs.get("taskGroupId")
+        if task_group_id:
+            filters.append(Q(taskgroup__user__id=task_group_id))
+
+        user = info.context.user
+        if type(user) is AnonymousUser:
+            raise PermissionError("Unauthorized")
```

Comment:
I think we can use 
```
        Query._check_permissions(
            info.context.user,
            TasksManagementConfig.gql_task_group_search_perms
        )
        ```
        as well, unless there's some case where TaskExecutor would be queried in other ways. For role quering we can rely on basic User query.